lib/ecto/postgres/pg_rational.ex

```diff
@@ -0,0 +1,157 @@
+use Vtc.Ecto.Postgres.Utils
+
+defpgmodule Vtc.Ecto.Postgres.PgRational do
+  @moduledoc """
+  Defines a composite type for storing rational values as dual int64s. These values
+  are cast to `%Ratio{}` structs for use in application code, provided by the `Ratio`
+  library.
+
+  The composite type is defined as follows:
+
+  ```sql
+  CREATE TYPE public.rational as (
+    numerator bigint,
+    denominator bigint
+  )
+  ```
+
+  Rational values can be cast in SQL expressions like so:
```

Comment:
i think i f'd up here grammatically, from my previous comment ...but was close hah.
```suggestion
  Rational values can be cast from SQL expressions like so:
```

Reply:
I think "in" is correct here. You are casting FROM an array, but doing so IN an SQL expression.

Reply:
interesting, not sure i agree -- but not going to split-hairs here, will leave the decision with you. `(1, 2)` is an "expression". you're then casting that expression into a `:rational` type. In that case, "from" makes more sense (since the expression is the originator).